WireDomain/Sources/WireDomain/Notifications/Components/PullEventsComponent.swift

```diff
@@ -0,0 +1,423 @@
+
+        return ProteusMessageDecryptor(
+            proteusService: proteusService,
+            userClientsLocalStore: userClientsLocalStore,
+            userLocalStore: userLocalStore
+        )
+    }
+
+    private var featureRepository: any FeatureRepositoryInterface {
+        FeatureRepository(context: dependency.coreData.syncContext)
+    }
+
+    private var updateEventsLocalStore: any UpdateEventsLocalStoreProtocol {
+        UpdateEventsLocalStore(
+            context: dependency.coreData.eventContext,
+            userID: dependency.userID,
+            sharedUserDefaults: dependency.sharedUserDefaults
+        )
+    }
+
+    private var userClientsLocalStore: any UserClientsLocalStoreProtocol {
+        UserClientsLocalStore(
+            context: dependency.coreData.syncContext
+        )
+    }
+
+    private func mlsMessageDecryptor(
+        selfUserID: UUID
+    ) -> any MLSMessageDecryptorProtocol {
+        let coreData = dependency.coreData
+        let coreCryptoProvider = coreCryptoProvider(
+            selfUserID: selfUserID
+        )
+
+        let commitSender = CommitSender(
+            coreCryptoProvider: coreCryptoProvider,
+            notificationContext: coreData.syncContext.notificationContext
+        )
+
+        let mlsActionExecutor = MLSActionExecutor(
+            coreCryptoProvider: coreCryptoProvider,
+            commitSender: commitSender,
+            featureRepository: featureRepository
+        )
+
+        let mlsDecryptionService = MLSDecryptionService(
+            context: coreData.syncContext,
+            mlsActionExecutor: mlsActionExecutor
+        )
+
+        return MLSMessageDecryptor(
+            mlsDecryptionService: mlsDecryptionService,
+            conversationLocalStore: conversationLocalStore
+        )
+    }
+
+    private var accountContainer: URL {
+        CoreDataStack.accountDataFolder(
+            accountIdentifier: dependency.userID,
+            applicationContainer: dependency.applicationContainer
+        )
+    }
+}
+
+extension PullEventsComponent {
+    func updateEventsAPI(
+        selfClientID: String
+    ) async -> any UpdateEventsAPI {
+        let authenticationManager = await makeAuthenticationManager(
+            selfClientID: selfClientID
+        )
+
+        let networkService = await makeNetworkService()
+
+        let apiService = APIService(
+            networkService: networkService,
+            authenticationManager: authenticationManager
+        )
+
+        let apiVersion = makeApiVersion()
+
+        return UpdateEventsAPIBuilder(
+            apiService: apiService
+        ).makeAPI(for: apiVersion)
+    }
+
+    func makeApiVersion() -> WireAPI.APIVersion {
+        let key = "SelectedAPIVersion"
+        let sharedUserDefaults = dependency.sharedUserDefaults
+
+        guard sharedUserDefaults.object(forKey: key) != nil else {
+            fatalError("API version not found")
```

Comment:
suggestion: we should log too, we can use the fatal method
```suggestion
            fatal("API version not found")
```

Reply:
done in https://github.com/wireapp/wire-ios/pull/2397/commits/233035022f1fe75ef5ce6043171fccccd94a38a7